### ISSUE TITLE
Steam: Core Dumping

### ISSUE THREAD

**Issue by mio-19** [FEX-Emu/FEX#2352]
**What Game**
Steam itself

**Describe the bug**
Core Dump

**To Reproduce**
1. install steam from official deb package:
```

# create necessary directories
mkdir -p ~/steam
mkdir -p ~/steam/tmp
cd ~/steam/tmp

# download latest deb and unpack
wget https://cdn.cloudflare.steamstatic.com/client/installer/steam.deb
ar x steam.deb
tar xf data.tar.xz

# remove deb archives, not needed anymore
rm ./*.tar.xz ./steam.deb

# move deb contents to steam folder
mv ./usr/* ../
cd ../ && rm -rf ./tmp/

# create run script
echo '#!/bin/bash
export STEAMOS=1
export STEAM_RUNTIME=1
export DBUS_FATAL_WARNINGS=0
exec FEXInterpreter ~/steam/bin/steam "$@"' > steam

# make script executable and move
chmod +x steam
mkdir -p ~/.local/bin
ln -s "$PWD/steam" ~/.local/bin/
```
2. `$ steam`

**Expected behavior**
Seeing UI

**Screenshots and Video**
```
$ steam
steam.sh[16930]: Running Steam on ubuntu 22.04 64-bit
steam.sh[16930]: STEAM_RUNTIME is enabled by the user
setup.sh[16995]: Steam runtime environment up-to-date!
steam.sh[16930]: Steam client's requirements are satisfied
[2023-01-26 02:24:06] Startup - updater built Dec 15 2022 21:26:49
[2023-01-26 02:24:06] Startup - Steam Client launched with: '/home/user/.local/share/Steam/ubuntu12_32/steam'
Installing breakpad exception handler for appid(steam)/version(1671236931)
Looks like steam didn't shutdown cleanly, scheduling immediate update check
[2023-01-26 02:24:07] Loading cached metrics from disk (/home/user/.local/share/Steam/package/steam_client_metrics.bin)
[2023-01-26 02:24:07] Failed to load cached hosts file (File 'update_hosts_cached.vdf' not found), using defaults
[2023-01-26 02:24:07] Using the following download hosts for Public, Realm steamglobal
[2023-01-26 02:24:07] 1. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
Installing breakpad exception handler for appid(steam)/version(1671236931)
[2023-01-26 02:24:07] Checking for update on startup
[2023-01-26 02:24:08] Checking for available updates...
[2023-01-26 02:24:08] Downloading manifest: http://media.steampowered.com/client/steam_client_ubuntu12
[2023-01-26 02:24:08] Manifest download: send request
Installing breakpad exception handler for appid(steam)/version(1671236931)
[2023-01-26 02:24:08] Manifest download: waiting for download to finish
[2023-01-26 02:24:08] Manifest download: finished
[2023-01-26 02:24:08] Download skipped: /client/steam_client_ubuntu12 version 1671236931, installed version 1671236931, existing pending version 0
[2023-01-26 02:24:08] Nothing to do
[2023-01-26 02:24:08] Verifying installation...
[2023-01-26 02:24:08] Performing checksum verification of executable files
[2023-01-26 02:24:10] Verification complete
Loaded SDL version 2.27.0-p7692409
Gtk-Message: 02:24:39.711: Failed to load module "gail"
Gtk-Message: 02:24:39.714: Failed to load module "atk-bridge"
XRRGetOutputInfo Workaround: initialized with override: 0 real: 0xecc479c0
XRRGetCrtcInfo Workaround: initialized with override: 0 real: 0xecc461f0

(steam:17058): Gtk-WARNING **: 02:24:39.801: Unable to locate theme engine in module_path: "adwaita",
/usr/share/themes/Yaru/gtk-2.0/main.rc:775: error: unexpected identifier 'direction', expected character '}'

(steam:17058): Gtk-WARNING **: 02:24:39.816: Unable to locate theme engine in module_path: "adwaita",
/usr/share/themes/Yaru/gtk-2.0/hacks.rc:28: error: invalid string constant "normal_entry", expected valid string constant
Fatal error: futex robust_list not initialized by pthreads
No minidump written, nothing to upload.
steamwebhelper.sh[17083]: Runtime for steamwebhelper: defaulting to /home/user/.local/share/Steam/ubuntu12_64/steam-runtime-heavy
steamwebhelper.sh[17083]: glibc >= 2.34, partially disabling sandbox until CEF supports clone3()
/home/user/.local/share/Steam/steam.sh: line 798: 17058 Segmentation fault      (core dumped) "$STEAMROOT/$STEAMEXEPATH" "$@"
```

**System information:**
 - OS: [eg: Ubuntu 21.10] Ubuntu 22.04
 - CPU/SoC: [eg: Snapdragon 888, Intel Core i8-12900k] Virtual machine on Apple M1
 - Video driver version: [eg: OpenGL ES 3.2 Mesa 22.0.0-devel (git-9ff086052a)]
 - RootFS used: [eg: Ubuntu 21.10 Official Rootfs] Ubuntu 22.04
 - FEX version: (FEXGetConfig --version) [eg: FEX-2112-155-gc691d709] FEX-2301
 - Thunks Enabled: [Yes/No]

**Additional context**
 - Is this an x86 or x86-64 game: [x86/x86-64/Both] x86
 - Does this reproduce on x86-64 host with FEX: [Yes/No/Untested] Untested
 - Does this reproduce on AArch64 with Radeon/Intel/Nvidia: [Yes/No/Untested] Untested
 - Is this a Vulkan game: [Yes/No/Unknown] No
   - If Yes, What is your Vulkan driver:

**Comment by PietHelzel** [comment]
I had the same problem on FEX-2301. I proceeded to use an older commit that works to start steam for the first time and logged in. Later I went back to 2301 and it also magically worked. So it seems like this crash only occurs on first startup, not on subsequent ones.

**Comment by Sonicadvance1** [comment]
As far as I can tell, this looks like this is a bug in the current PPA release. Rebuilding 2301 locally doesn't reproduce but installing the PPA version does.
Not sure what managed to break there.

**Comment by Sonicadvance1** [comment]
This /might/ be fixed with https://github.com/FEX-Emu/FEX/pull/2420 merged. There was a nasty race condition which manifested similarly.

**Comment by Sonicadvance1** [comment]
Should be fixed by #2714
